grakn-graql/src/test/java/ai/grakn/graql/internal/analytics/CorenessTest.java

```diff
@@ -0,0 +1,349 @@
+
+        addSchemaAndEntities();
+
+        List<Long> list = new ArrayList<>(4);
+        long workerNumber = 4L;
+        for (long i = 0L; i < workerNumber; i++) {
+            list.add(i);
+        }
+
+        Set<Map<Integer, Set<String>>> result = list.parallelStream().map(i -> {
+            try (GraknTx graph = session.open(GraknTxType.READ)) {
+                return Graql.compute().withTx(graph).coreness().minK(3).execute();
+            }
+        }).collect(Collectors.toSet());
+        assertEquals(1, result.size());
+        result.forEach(map -> {
+            assertEquals(1, map.size());
+            assertEquals(4, map.get(3).size());
+        });
+    }
+
+    private void addSchemaAndEntities() throws InvalidKBException {
+        try (GraknTx graph = session.open(GraknTxType.WRITE)) {
+            EntityType entityType1 = graph.putEntityType(thing);
+            EntityType entityType2 = graph.putEntityType(anotherThing);
+
+            Role role1 = graph.putRole("role1");
+            Role role2 = graph.putRole("role2");
+            RelationshipType relationshipType1 = graph.putRelationshipType(related)
+                    .relates(role1).relates(role2);
+
+            Role role3 = graph.putRole("role3");
+            Role role4 = graph.putRole("role4");
+            RelationshipType relationshipType2 = graph.putRelationshipType(veryRelated)
+                    .relates(role3).relates(role4);
+
+            entityType1.plays(role1).plays(role2).plays(role3).plays(role4);
+            entityType2.plays(role1).plays(role2).plays(role3).plays(role4);
+
+            Entity entity1 = entityType1.addEntity();
+            Entity entity2 = entityType1.addEntity();
+            Entity entity3 = entityType2.addEntity();
+            Entity entity4 = entityType2.addEntity();
+
+            relationshipType1.addRelationship()
+                    .addRolePlayer(role1, entity1)
+                    .addRolePlayer(role2, entity2);
+            relationshipType1.addRelationship()
+                    .addRolePlayer(role1, entity2)
+                    .addRolePlayer(role2, entity3);
+            relationshipType1.addRelationship()
+                    .addRolePlayer(role1, entity3)
+                    .addRolePlayer(role2, entity4);
+            relationshipType1.addRelationship()
+                    .addRolePlayer(role1, entity4)
+                    .addRolePlayer(role2, entity1);
+
+            relationshipType2.addRelationship()
+                    .addRolePlayer(role3, entity1)
+                    .addRolePlayer(role4, entity3);
+            relationshipType2.addRelationship()
+                    .addRolePlayer(role3, entity2)
+                    .addRolePlayer(role4, entity4);
+
+            entityId1 = entity1.getId();
+            entityId2 = entity2.getId();
+            entityId3 = entity3.getId();
+            entityId4 = entity4.getId();
+
+            graph.commit();
+        }
+    }
+}
```

Comment:
Might be worth describing the configuration graphically, saves time when dealing with test failures.

Reply:
That's a good idea man. I will try to draw this in the test, something like this:
```/*
        root    10
    	(0) -------> (1) \
    	 |  \       /  ^  \
    	 |   \30   |   |20 \
    	 |10  \    |10 |    \10
    	 |     \   |  /      \
    	 V  15  V  V /   20   V
    	(3)<----- (2) -----> (4)
    	  \-------^
    	     40
    	 */